app/controllers/users_controller.rb

```diff
@@ -22,11 +25,33 @@ def create
     end
   end
 
+  def edit
+    @user = User.find_by id: params[:id]
+  end
+
+  def update
+    @user = User.find_by id: params[:id]
+    if @user.update_attributes(user_params)
+      flash[:success] = t ".flash_update"
+      redirect_to root_path
+    else
+      render :edit
+    end
+  end
+
   private
 
   def user_params
     params.require(:user).permit :name, :email, :password,
                                  :password_confirmation, :phone, :gender,
                                  :birthday, :city
   end
+
+  def logged_in_user
+      unless logged_in?
```

Comment:
sai indent và suggest code theo hướng sau
```suggestion
    return if logged_in?
    flash[:danger] = t ".flash_login"
    redirect_to login_url
```